File: trab_phyton/final_phyton/com.example.final_project_fxml/MainController.py
import tkinter as tk
from tkinter import ttk
from tkinter import messagebox
from hostelapp.model import Address, Hostel
import pickle  

class ControladorPrincipal:
    def __init__(self, root):
        self.root = root
        self.root.title("Projeto Final")

        self.hostel_name = tk.StringVar()
        self.desc_label_text = tk.StringVar()
        self.end_label_text = tk.StringVar()
        self.date_label_text = tk.StringVar()

        self.hostel = self.read_hostel_from_file()

        self.create_widgets()

    def create_widgets(self):

        tk.Label(self.root, text="Nome do Albergue:").grid(row=0, column=0)
        tk.Label(self.root, textvariable=self.hostel_name).grid(row=0, column=1)

        tk.Label(self.root, text="Descrição:").grid(row=1, column=0)
        desc_label = tk.Label(self.root, textvariable=self.desc_label_text, wraplength=300)
        desc_label.grid(row=1, column=1)

        tk.Label(self.root, text="Endereço:").grid(row=2, column=0)
        tk.Label(self.root, textvariable=self.end_label_text).grid(row=2, column=1)

        tk.Label(self.root, text="Data de Inauguração:").grid(row=3, column=0)
        tk.Label(self.root, textvariable=self.date_label_text).grid(row=3, column=1)


        tk.Button(self.root, text="Cadastro de Hóspede", command=self.open_guest_creation).grid(row=4, column=0)
        tk.Button(self.root, text="Lista de Hóspedes", command=self.open_guest_list).grid(row=4, column=1)
        tk.Button(self.root, text="Informações do Hostel", command=self.open_hostel_info).grid(row=5, column=0)
        tk.Button(self.root, text="Cadastro de Quarto", command=self.open_room_creation).grid(row=5, column=1)

        self.atualizar_info()

    def atualizar_info(self):
        if self.hostel is not None:
            self.hostel_name.set(self.hostel.getName())
            self.desc_label_text.set(self.hostel.getDescription())
            endereco = self.hostel.getAddress()
            end_str = f"Endereco: {endereco.getAddress()}\nZip-Code: {endereco.getZipCode()}\nCidade: {endereco.getCity()}\nEstado: {endereco.getState()}\nPais: {endereco.getCountry()}"
            self.end_label_text.set(end_str)
            data = self.hostel.getInaugurationDate().toString()
            self.date_label_text.set("Desde: " + data)


    def read_hostel_from_file(self):
        try:
            with open("hostel_info.dat", "rb") as file:
                return pickle.load(file)
        except (IOError, pickle.UnpicklingError) as e:
            print("Erro ao ler as informações do albergue do arquivo:", e)
            return None

if __name__ == "__main__":
    root = tk.Tk()
    app = ControladorPrincipal(root)
    root.mainloop()

File: trab_java/final_project/src/main/java/com/example/final_project/MainController.java
package com.example.final_project;

import hostelapp.model.Address;
import hostelapp.model.Hostel;
import javafx.fxml.FXML;
import javafx.scene.control.Label;
import javafx.event.ActionEvent;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.ObjectInputStream;

public class MainController{

    private final String FILE_PATH = "hostel_info.dat";

    @FXML private Label hostel_name;
    @FXML private Label desc_label;
    @FXML private Label end_label;
    @FXML private Label date_label;

    private Hostel hostel;

    @FXML
    public void initialize() {
        hostel = readHostelFromFile();
        if (hostel != null) {
            hostel_name.setText(hostel.getName());
            desc_label.setWrapText(true);
            desc_label.setText(hostel.getDescription());
            end_label.setText(toString(hostel.getAddress()));
            String data = hostel.getInaugurationDate().toString();
            date_label.setText("Desde: " + data);
        }
    }


    public void atulizar_info(){
        Hostel hostel = Hostel.getHostel();
        hostel_name.setText(hostel.getName());
        desc_label.setWrapText(true);
        desc_label.setText(hostel.getDescription());
        end_label.setText(toString(hostel.getAddress()));
        String data = hostel.getInaugurationDate().toString();
        date_label.setText("Desde: " + data);
    }

    private String toString(Address address) {
        return "Endereco: " + address.getAddress() + "\nZip-Code:" + address.getZipCode()
                + "\nCidade: " + address.getCity() + "\nEstado: " + address.getState()
                + "\nPais: " + address.getCountry();
    }

    public void openGuestCreation (ActionEvent event) throws Exception {
        Parent newInterfaceView = FXMLLoader.load(getClass().getResource("guest_creation.fxml"));

        Stage newInterfaceStage = new Stage();
        newInterfaceStage.setTitle("Cadatro de Hospede");
        newInterfaceStage.setScene(new Scene(newInterfaceView));
        newInterfaceStage.show();
    }
    public void openGuestList (ActionEvent event) throws Exception {
        Parent newInterfaceView = FXMLLoader.load(getClass().getResource("guest_list.fxml"));

        Stage newInterfaceStage = new Stage();
        newInterfaceStage.setTitle("Lista de Hospedes");
        newInterfaceStage.setScene(new Scene(newInterfaceView));
        newInterfaceStage.show();
    }
    public void openHostelInfo (ActionEvent event) throws Exception {
        Parent newInterfaceView = FXMLLoader.load(getClass().getResource("hostel_info.fxml"));

        Stage newInterfaceStage = new Stage();
        newInterfaceStage.setTitle("Informações do Hostel");
        newInterfaceStage.setScene(new Scene(newInterfaceView));
        newInterfaceStage.show();
    }
    public void openRoomCreation (ActionEvent event) throws Exception {
        Parent newInterfaceView = FXMLLoader.load(getClass().getResource("room_creation.fxml"));

        Stage newInterfaceStage = new Stage();
        newInterfaceStage.setTitle("Cadatro de Quarto");
        newInterfaceStage.setScene(new Scene(newInterfaceView));
        newInterfaceStage.show();
    }

        public Hostel readHostelFromFile() {
            try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(FILE_PATH))) {
                return (Hostel) ois.readObject();
            } catch (IOException | ClassNotFoundException e) {
                System.err.println("Error while reading hostel info from file: " + e.getMessage());
                return null;
            }
        }
}